dev/lib/product_taxonomy/commands/sync_en_localizations_command.rb

```diff
@@ -41,13 +41,74 @@ def sync_values
       write_localizations("values", localizations)
     end
 
+    # Writes localization data to a YAML file with 'context' keys converted to comments.
+    # Uses YAML.dump for formatting, then injects context as comments above entry fields.
     def write_localizations(type, localizations)
       file_path = File.expand_path("localizations/#{type}/en.yml", ProductTaxonomy.data_path)
+
+      context_map = extract_contexts(localizations)
+
+      yaml_output = YAML.dump(localizations, line_width: -1)
+      yaml_with_comments = inject_context_comments(yaml_output, context_map, type)
+
       File.open(file_path, "w") do |file|
         file.puts "# This file is auto-generated. Do not edit directly."
-        file.write(YAML.dump(localizations, line_width: -1))
+        file.write(yaml_with_comments)
       end
       logger.info("Wrote #{type} localizations to #{file_path}")
     end
+
+    # Extracts and removes 'context' keys from the localization hash.
+    #
+    # Expects a specific structure from LocalizationsSerializer:
+    #   { "en" => { "section" => { "entry-id" => { "name" => "...", "context" => "..." } } } }
+    #
+    # Note: This creates a flat map keyed by entry ID only, so it assumes entry IDs are unique
+    # across all sections. In practice, each file (categories/attributes/values) contains only
+    # one section, so there are no collisions.
```

Comment:
Worth a `raise` if a given id already exists in the map when we process it in case any assumptions change here?